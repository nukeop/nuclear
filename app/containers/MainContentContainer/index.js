import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as Actions from '../../actions';

import MainLayout from '../../components/MainLayout';

import AlbumViewContainer from '../AlbumViewContainer';
import ArtistViewContainer from '../ArtistViewContainer';
import DashboardContainer from '../DashboardContainer';
import FavoritesContainer from '../FavoritesContainer';
import LyricsContainer from '../LyricsContainer';
import PlaylistsContainer from '../PlaylistsContainer';
import PlaylistViewContainer from '../PlaylistViewContainer';
import PluginsContainer from '../PluginsContainer';
import SearchResultsContainer from '../SearchResultsContainer';
import SettingsContainer from '../SettingsContainer';
import TagViewContainer from '../TagViewContainer';
import EqualizerViewContainer from '../EqualizerViewContainer';
import LibraryViewContainer from '../LibraryViewContainer';

import DownloadsContainer from '../../containers/DownloadsContainer';

class MainContentContainer extends React.Component {
  componentDidMount () {
    if (this.props.history &&
      this.props.location &&
      this.props.location.pathname === '/') {
      this.props.history.push('/dashboard');
    }
  }

  render () {
    return (
      <Route render={({ location }) => {
        return (
          <MainLayout>
            <Switch key={location.key} location={location}>
              <Route path='/album/:albumId' component={AlbumViewContainer} />
              <Route path='/artist/:artistId' component={ArtistViewContainer} />
              <Route path='/dashboard' component={DashboardContainer} />
              <Route path='/downloads' component={DownloadsContainer} />
              <Route path='/favorites/tracks' component={FavoritesContainer} />
              <Route path='/playlists' component={PlaylistsContainer} />
              <Route path='/playlist/:playlistId' component={PlaylistViewContainer} />
              <Route path='/plugins' component={PluginsContainer} />
              <Route path='/settings' component={SettingsContainer} />
              <Route path='/tag/:tagName' component={TagViewContainer} />
              <Route path='/search' component={SearchResultsContainer} />
              <Route path='/lyrics' component={LyricsContainer} />
              <Route path='/equalizer' component={EqualizerViewContainer} />
              <Route path='/library' component={LibraryViewContainer} />
            </Switch>
          </MainLayout>
        );
      }
      } />
    );
  }
}

function mapStateToProps () {
  return {};
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators(Actions, dispatch)
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MainContentContainer));
