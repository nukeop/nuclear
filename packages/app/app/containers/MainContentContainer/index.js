import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as SearchActions from '../../actions/search';

import MainLayout from '../../components/MainLayout';

import AlbumViewContainer from '../AlbumViewContainer';
import ArtistViewContainer from '../ArtistViewContainer';
import DashboardContainer from '../DashboardContainer';
import DownloadsContainer from '../DownloadsContainer';
import EqualizerViewContainer from '../EqualizerViewContainer';
import FavoritesContainer from '../FavoritesContainer';
import LibraryViewContainer from '../LibraryViewContainer';
import LyricsContainer from '../LyricsContainer';
import PlaylistViewContainer from '../PlaylistViewContainer';
import PlaylistsContainer from '../PlaylistsContainer';
import PluginsContainer from '../PluginsContainer';
import SearchResultsContainer from '../SearchResultsContainer';
import SettingsContainer from '../SettingsContainer';
import TagViewContainer from '../TagViewContainer';
import VisualizerNode from '../VisualizerContainer/VisualizerNode';

class MainContentContainer extends React.Component {
  componentDidMount () {
    if (this.props.history &&
      this.props.location &&
      this.props.location.pathname === '/') {
      this.props.history.push('/dashboard');
    }
  }

  render () {
    return (
      <Route render={({ location }) => {
        return (
          <MainLayout>
            <Switch key={location.key} location={location}>
              <Route path='/album/:albumId' component={AlbumViewContainer} />
              <Route path='/artist/:artistId' component={ArtistViewContainer} />
              <Route path='/dashboard' component={DashboardContainer} />
              <Route path='/downloads' component={DownloadsContainer} />
              <Route path='/favorites/albums' component={FavoritesContainer} />
              <Route path='/favorites/tracks' component={FavoritesContainer} />
              <Route path='/favorites/artists' component={FavoritesContainer} />
              <Route path='/playlists' component={PlaylistsContainer} />
              <Route path='/playlist/:playlistId' component={PlaylistViewContainer} />
              <Route path='/plugins' component={PluginsContainer} />
              <Route path='/settings' component={SettingsContainer} />
              <Route path='/tag/:tagName' component={TagViewContainer} />
              <Route path='/search' component={SearchResultsContainer} />
              <Route path='/lyrics' component={LyricsContainer} />
              <Route path='/equalizer' component={EqualizerViewContainer} />
              <Route path='/visualizer' component={VisualizerNode} />
              <Route path='/library' component={LibraryViewContainer} />
            </Switch>
          </MainLayout>
        );
      }
      } />
    );
  }
}

function mapStateToProps () {
  return {};
}

function mapDispatchToProps (dispatch) {
  return {
    searchActions: bindActionCreators(SearchActions, dispatch)
  };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MainContentContainer));
