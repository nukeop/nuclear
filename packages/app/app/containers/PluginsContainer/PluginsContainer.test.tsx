import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import { buildStoreState } from '../../../test/storeBuilders';
import { AnyProps, configureMockStore, setupI18Next, TestRouterProvider, TestStoreProvider } from '../../../test/testUtils';
import MainContentContainer from '../MainContentContainer';

jest.mock('fs');

describe('Plugins container', () => {
  beforeAll(() => {
    setupI18Next();
  });

  it('should display plugins', () => {
    const { component } = mountComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should be able to change the streaming provider plugin', async () => {
    const { component, store } = mountComponent();
    await waitFor(() => component.getAllByText(/Test Stream Provider/i)[0].click());
    await waitFor(() => component.getByText(/Different Stream Provider/i).click());

    const state = store.getState();
    expect(state.plugin.selected.streamProviders).toEqual('Different Stream Provider');
  });

  it('should be able to change the metadata provider plugin', async () => {
    const { component, store } = mountComponent();
    await waitFor(() => component.getAllByText(/Test Meta Provider/i)[0].click());
    await waitFor(() => component.getByText(/Another Meta Provider/i).click());

    const state = store.getState();
    expect(state.plugin.selected.metaProviders).toEqual('Another Metadata Provider');
  });

  it('should be able to change the lyrics provider plugin', async () => {
    const { component, store } = mountComponent();
    await waitFor(() => component.getAllByText(/Test Lyrics Provider/i)[0].click());
    await waitFor(() => component.getByText(/Different Lyrics Provider/i).click());

    const state = store.getState();
    expect(state.plugin.selected.lyricsProviders).toEqual('Different Lyrics Provider');
  });

  it('should be able to load a user plugin', async () => {
    const { component, store } = mountComponent();
    await waitFor(() => component.getAllByText(/Add a plugin/i)[0].click());

    const state = store.getState();

    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const remote = require('electron').remote;
    expect(remote.dialog.showOpenDialog).toHaveBeenCalledWith({
      filters: [{
        name: 'Javascript files',
        extensions: ['js', 'jsx']
      }]
    });
    expect(state.plugin.userPlugins['test file.txt']).toEqual(
      expect.objectContaining({
        path: 'test file.txt'
      })
    );
  });

  it('should display loaded user plugins', async () => {
    const state = buildStoreState()
      .withPlugins()
      .build();

    state.plugin = {
      ...state.plugin,
      userPlugins: {
        'test file.txt': {
          path: 'test file.txt',
          name: 'test plugin',
          description: 'test plugin description',
          image: null
        }
      }
    };

    const { component } = mountComponent(state);

    expect(component.asFragment()).toMatchSnapshot();
  });

  const mountComponent = (initialStore?: AnyProps) => {
    const initialState = initialStore ||
      buildStoreState()
        .withPlugins()
        .build();

    const history = createMemoryHistory({
      initialEntries: ['/plugins']
    });

    const store = configureMockStore(initialState);

    const component = render(
      <TestRouterProvider
        history={history}
      >
        <TestStoreProvider
          store={store}
        >
          <MainContentContainer />
        </TestStoreProvider>
      </TestRouterProvider >
    );

    return { component, history, store };
  };
});
