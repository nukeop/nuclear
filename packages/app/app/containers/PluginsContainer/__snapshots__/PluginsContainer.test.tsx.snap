// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugins container should display loaded user plugins 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="plugins_view_container"
    >
      <section
        class="plugin_settings_section"
      >
        <div
          class="header_container"
        >
          Stream providers
        </div>
        <hr />
        <div
          class="ui segment"
        >
          <label>
            Select the default music source:
          </label>
          <div
            aria-expanded="false"
            class="ui selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="divider text"
              role="alert"
            >
              Test Stream Provider
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                aria-checked="true"
                aria-selected="true"
                class="active selected item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Test Stream Provider
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Different Stream Provider
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="plugin_settings_section"
      >
        <div
          class="header_container"
        >
          Metadata providers
        </div>
        <hr />
        <div
          class="ui segment"
        >
          <label>
            Select the default metadata provider:
          </label>
          <div
            aria-expanded="false"
            class="ui selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="divider text"
              role="alert"
            >
              Test Meta Provider
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                aria-checked="true"
                aria-selected="true"
                class="active selected item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Test Meta Provider
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Another Meta Provider
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="plugin_settings_section"
      >
        <div
          class="header_container"
        >
          Lyrics providers
        </div>
        <hr />
        <div
          class="ui segment"
        >
          <label>
            Select the default lyrics provider:
          </label>
          <div
            aria-expanded="false"
            class="ui selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="divider text"
              role="alert"
            >
              Test Lyrics Provider
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                aria-checked="true"
                aria-selected="true"
                class="active selected item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Test Lyrics Provider
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Different Lyrics Provider
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="plugin_settings_section"
      >
        <div
          class="header_container"
        >
          User plugins
        </div>
        <hr />
        <div
          class="ui segment user_plugins_section"
        >
          <div
            class="ui warning message plugins_warning"
          >
            <div
              class="header"
            >
              <i
                aria-hidden="true"
                class="warning sign icon"
              />
              Danger zone!
            </div>
            <p>
              Plugins work by running code on your computer. Load plugins only from sources you trust!
            </p>
          </div>
          <hr />
          <button
            class="ui icon inverted button"
          >
            <i
              aria-hidden="true"
              class="plus icon"
            />
            Add a plugin
          </button>
          <div
            class="nuclear user_plugins_item"
          >
            <div
              class="plugin_info"
            >
              <div
                class="plugin_name"
              >
                test plugin
              </div>
              <div
                class="plugin_path"
              >
                <label>
                  Installed from:
                </label>
                <span>
                  test file.txt
                </span>
              </div>
              <div
                class="plugin_description"
              >
                test plugin description
              </div>
              <div
                class="plugin_footer"
              >
                <div
                  class="plugin_author"
                >
                  <label>
                    Author:
                  </label>
                  <span
                    class="link"
                  />
                </div>
                <button
                  class="ui basic icon inverted button delete_button"
                >
                  <i
                    aria-hidden="true"
                    class="trash alternate outline icon"
                  />
                  Uninstall
                </button>
              </div>
              <div
                class="error_footer"
              />
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Plugins container should display plugins 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="plugins_view_container"
    >
      <section
        class="plugin_settings_section"
      >
        <div
          class="header_container"
        >
          Stream providers
        </div>
        <hr />
        <div
          class="ui segment"
        >
          <label>
            Select the default music source:
          </label>
          <div
            aria-expanded="false"
            class="ui selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="divider text"
              role="alert"
            >
              Test Stream Provider
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                aria-checked="true"
                aria-selected="true"
                class="active selected item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Test Stream Provider
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Different Stream Provider
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="plugin_settings_section"
      >
        <div
          class="header_container"
        >
          Metadata providers
        </div>
        <hr />
        <div
          class="ui segment"
        >
          <label>
            Select the default metadata provider:
          </label>
          <div
            aria-expanded="false"
            class="ui selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="divider text"
              role="alert"
            >
              Test Meta Provider
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                aria-checked="true"
                aria-selected="true"
                class="active selected item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Test Meta Provider
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Another Meta Provider
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="plugin_settings_section"
      >
        <div
          class="header_container"
        >
          Lyrics providers
        </div>
        <hr />
        <div
          class="ui segment"
        >
          <label>
            Select the default lyrics provider:
          </label>
          <div
            aria-expanded="false"
            class="ui selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="divider text"
              role="alert"
            >
              Test Lyrics Provider
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                aria-checked="true"
                aria-selected="true"
                class="active selected item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Test Lyrics Provider
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events: all;"
              >
                <span
                  class="text"
                >
                  Different Lyrics Provider
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="plugin_settings_section"
      >
        <div
          class="header_container"
        >
          User plugins
        </div>
        <hr />
        <div
          class="ui segment user_plugins_section"
        >
          <div
            class="ui warning message plugins_warning"
          >
            <div
              class="header"
            >
              <i
                aria-hidden="true"
                class="warning sign icon"
              />
              Danger zone!
            </div>
            <p>
              Plugins work by running code on your computer. Load plugins only from sources you trust!
            </p>
          </div>
          <hr />
          <button
            class="ui icon inverted button"
          >
            <i
              aria-hidden="true"
              class="plus icon"
            />
            Add a plugin
          </button>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
